controlplane/kubeadm/internal/control_plane.go

```diff
@@ -115,13 +124,12 @@ func (c *ControlPlane) FailureDomainWithMostMachines(machines FilterableMachineC
 	return PickMost(c, machines)
 }
 
-// FailureDomainWithFewestMachines returns the failure domain with the fewest number of machines.
-// Used when scaling up.
-func (c *ControlPlane) FailureDomainWithFewestMachines() *string {
+// NextFailureDomainForScaleUp returns the failure domain with the fewest number of up-to-date machines.
+func (c *ControlPlane) NextFailureDomainForScaleUp() *string {
```

Comment:
This rename is just because it would be unwieldy to change this to "FailureDomainWithFewestUpToDateMachines" 😸 . I think it's better to name it for what it represents rather than what it does.